src/main/kotlin/no/digdir/informasjonsforvaltning/fdk_dataservice_harvester/configuration/SecurityConfig.kt

```diff
@@ -0,0 +1,46 @@
+package no.digdir.informasjonsforvaltning.fdk_dataservice_harvester.configuration
+
+import org.springframework.boot.autoconfigure.security.oauth2.resource.OAuth2ResourceServerProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.HttpMethod
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.oauth2.core.DelegatingOAuth2TokenValidator
+import org.springframework.security.oauth2.jwt.*
+import org.springframework.security.oauth2.jwt.JwtClaimNames.AUD
+
+@Configuration
+open class SecurityConfig : WebSecurityConfigurerAdapter() {
+    override fun configure(http: HttpSecurity) {
+        http.csrf().disable()
+
+        http.cors()
+            .and()
+                .authorizeRequests()
+                    .antMatchers(HttpMethod.OPTIONS)
+                        .permitAll()
+                    .antMatchers(HttpMethod.POST, "/update/meta")
+                        .authenticated()
+                    .antMatchers(HttpMethod.GET)
+                        .permitAll()
+                    .anyRequest()
+                        .authenticated()
+            .and()
+                .oauth2ResourceServer()
+                    .jwt()
+    }
+
+    @Bean
+    open fun jwtDecoder(properties: OAuth2ResourceServerProperties): JwtDecoder? {
+        val jwtDecoder = NimbusJwtDecoder.withJwkSetUri(properties.jwt.jwkSetUri).build()
+        jwtDecoder.setJwtValidator(
+                DelegatingOAuth2TokenValidator(
+                    JwtTimestampValidator(),
+                    JwtIssuerValidator(properties.jwt.issuerUri),
+                    JwtClaimValidator(AUD) { aud: List<String> -> aud.contains("fdk-harvest-admin") }
```

Comment:
Stemmer det med `fdk-harvest-admin`?

Reply:
Vi bruker token som blir generert for harvest-admin når vi bruker endepunktet. I tillegg så regner jeg med at det er i harvest-admin vi legger det til om vi noen gang lager gui for dette, så bruker samme audience.

Alternativet er at vi legger til egne audience-felt for høsterene, men vi kommer til å legge det til i harvest-admin, så det føles veldig overflødig.